pkg/registry/client.go

```diff
@@ -68,11 +75,13 @@ type (
 		username           string
 		password           string
 		out                io.Writer
-		authorizer         auth.Client
+		authorizer         *auth.Client
 		registryAuthorizer RemoteClient
+		credentialsStore   credentials.Store
 		resolver           func(ref registry.Reference) (remotes.Resolver, error)
```

Comment:
If I'm not mistaken, this never gets assigned and will probably cause a nil pointer dereference in `Client.Resolve` on line 837:

https://github.com/helm/helm/blob/6505d416f618be17a481beb19b344369be141daa/pkg/registry/client.go#L827-L844

Reply:
@twz123  It appears to be set in `NewClient()` on lines 147 or 150, depending on dockerStore status.

Reply:
@scottrigby I wasn't talking about `credentialsStore` in line 80, but about `resolver` in line 81. This is not directly obvious from how GitHub displays this comment. Sorry, I should have been clearer.

Still pretty sure that `resolver` never gets assigned and will produce a nil pointer dereference in the aforementioned code block.

Reply:
@twz123 that makes more sense. Not needed for Helm 4, but to continue attempting a reasonable backwards compatibility for Helm 3 (re this thread https://github.com/helm/helm/pull/13382#discussion_r1836977600), we'd need to keep `resolver` in the struct. For that, what would you recommend?

Reply:
I removed `resolver` in the latest changes. It is no longer needed.

Reply:
`resolver` was private, so I don't think we need to keep it in the struct, but the latest changes keeps ClientOptResolver with a deprecated message.